webpack/assets/javascripts/react_app/components/SearchBar/SearchAutocomplete.js

```diff
@@ -0,0 +1,166 @@
+  const searchWrapperRef = useRef(null);
+  const searchInputRef = useRef(null);
+  const autocompleteRef = useRef(null);
+
+  const _onSearch = searchValue => {
+    setIsAutocompleteOpen(false);
+    onSearch(searchValue);
+  };
+  const onClear = () => {
+    onSearchChange('');
+  };
+
+  const onChange = (newValue, e) => {
+    if (
+      searchInputRef &&
+      searchInputRef.current &&
+      searchInputRef.current.contains(document.activeElement)
+    ) {
+      setIsAutocompleteOpen(true);
+      onSearchChange(newValue);
+    }
+  };
+
+  const onSelect = (e, itemId) => {
+    e.stopPropagation();
+    if (itemId[itemId.length - 1] !== ' ') {
+      itemId = `${itemId} `;
+    }
+    if (itemId[0] === ' ') {
+      itemId = itemId.slice(1);
+    }
+    onSearchChange(itemId);
+    searchInputRef.current.focus();
+  };
+
+  const handleMenuKeys = event => {
+    // keyboard shortcut to focus the search, will not focus if the key is typed into an input
+    if (event.key === '/' && event.target.tagName !== 'INPUT') {
+      event.preventDefault();
+      searchInputRef.current.focus();
+    }
+    // if the autocomplete is open and the browser focus is on the search input,
+    else if (
+      isAutocompleteOpen &&
+      searchInputRef.current &&
+      searchInputRef.current === event.target
+    ) {
+      // the escape key closes the autocomplete menu and keeps the focus on the search input.
+      if (event.key === 'Escape') {
+        setIsAutocompleteOpen(false);
+        searchInputRef.current.focus();
+        // the up and down arrow keys move browser focus into the autocomplete menu
+      } else if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+        const firstElement = autocompleteRef.current.querySelector(
+          'li > button:not(:disabled)'
+        );
+        firstElement && firstElement.focus();
+        event.preventDefault(); // by default, the up and down arrow keys scroll the window
+      } else if (
+        isAutocompleteOpen &&
+        autocompleteRef?.current?.contains(event.target) &&
+        event.key === 'Tab'
+      ) {
+        event.preventDefault();
+        setIsAutocompleteOpen(false);
+        searchInputRef.current.focus();
+      }
+    }
+  };
+
+  // The autocomplete menu should close if the user clicks outside the menu.
+  const handleClickOutside = event => {
+    if (
+      isAutocompleteOpen &&
+      searchWrapperRef &&
+      searchWrapperRef.current &&
+      !searchWrapperRef.current.contains(event.target)
+    ) {
+      setIsAutocompleteOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('keydown', handleMenuKeys);
+    window.addEventListener('click', handleClickOutside);
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClickOutside);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Please find a way to make it work without disabling the rule :angel:

Reply:
moved the functions to the useeffect